src/matrices.jl

```diff
@@ -155,22 +156,25 @@ by the following keyword arguments:
   and `scale` is ignored.
 * `metric="euclidean"` : metric of the distances, either `Metric` or a string,
    as in [`distancematrix`](@ref).
+* `parallel=false` : whether to parallelize the computation of the recurrence
+   matrix.  This will split the computation of the matrix across multiple threads.
+   This **may not work on Julia versions before v1.3**, so be warned!
 
 See also: [`CrossRecurrenceMatrix`](@ref), [`JointRecurrenceMatrix`](@ref) and
 use [`recurrenceplot`](@ref) to turn the result of these functions into a plottable format.
 
 ## References
-[1] : N. Marwan *et al.*, "Recurrence plots for the analysis of complex systems",
+[^1] : N. Marwan *et al.*, "Recurrence plots for the analysis of complex systems",
 *Phys. Reports 438*(5-6), 237-329 (2007).
 
-[2] : N. Marwan & C.L. Webber, "Mathematical and computational foundations of
+[^2] : N. Marwan & C.L. Webber, "Mathematical and computational foundations of
 recurrence quantifications", in: Webber, C.L. & N. Marwan (eds.), *Recurrence
 Quantification Analysis. Theory and Best Practices*, Springer, pp. 3-43 (2015).
 """
-function RecurrenceMatrix(x, ε; metric = DEFAULT_METRIC, kwargs...)
+function RecurrenceMatrix(x, ε; metric = DEFAULT_METRIC, parallel::Bool = false, kwargs...)
     m = getmetric(metric)
     s = resolve_scale(x, m, ε; kwargs...)
-    m = recurrence_matrix(x, m, s)
+    m = recurrence_matrix(x, m, s, Val(parallel))
```

Comment:
Does using `Val` result into performance gain, given the cool optimizations of the latest Julia version? I know it makes it "more type stable", but is it actually worth it? (because it sacrifices code clarity)

Reply:
I actually think it increases code clarity, which is why I used it - we can separate implementation methods.  Later, if we choose to define a distributed or GPU backend for calculating recurrence matrices, it will become easier to implement if we're already dispatching on types as opposed to having an if statement with keyword arguments.

Reply:
Hm, okay sounds convincing.